libs/wingc/src/type_check.rs

```diff
@@ -794,7 +861,13 @@ impl Display for Type {
 			Type::Duration => write!(f, "duration"),
 			Type::Boolean => write!(f, "bool"),
 			Type::Void => write!(f, "void"),
-			Type::Json => write!(f, "Json"),
+			Type::Json(t) => {
+				if let Some(t) = t {
+					write!(f, "Json {}", t)
+				} else {
+					write!(f, "Json")
+				}
+			}
```

Comment:
I'm not sure we should be showing the inferred Json literal information publicly as part of the types.

<img width="623" alt="Screenshot 2023-08-11 at 12 59 40 PM" src="https://github.com/winglang/wing/assets/5008987/e62bdf2d-6112-4346-9bd9-3d6143924e50">

It looks like this info appears in the LSP. IMO (at least initially) this info should just be information we're storing under the hood, and not something leaks into the user's mental model of the Wing type system.

Here's another example where this leads to some confusing behavior:

<img width="409" alt="Screenshot 2023-08-11 at 1 07 45 PM" src="https://github.com/winglang/wing/assets/5008987/358c0793-1e40-447b-a1a3-cb5dcc281ee5">

Reply:
> this info should just be information we're storing under the hood, and not something leaks into the user's mental model of the Wing type system.

Not sure I agree, I think the user's mental model could handle the extra information. I will go ahead and remove it for now though.


> Here's another example where this leads to some confusing behavior:

This is actually a bug. I think I was too optimistic by letting the return statement set that. Will fix.

Reply:
As a rule of thumb, I think most [1] type annotations a user gets on hover should be annotations that they can type into their code - it makes for a more consistent experience. Showing the inferred Json type would be an issue since this isn't supported in Wing:

```js
let x: Json { base: str } = { base: "d" };
```

[1] Exceptions in partially-compiled programs like "unknown" are probably OK